cmds/chroot/chroot.go

```diff
@@ -0,0 +1,206 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+var (
+	userspecFlag  string
+	groupsFlag    string
+	skipchdirFlag bool
+)
+
+func init() {
+	flag.StringVar(&userspecFlag, "u", "", "specify user and group (ID only) as USER:GROUP")
+	flag.StringVar(&groupsFlag, "g", "", "specify supplementary group ids as g1,g2,..,gN")
+	flag.BoolVar(&skipchdirFlag, "s", false, fmt.Sprint("Use this option to not change",
+		"the working directory to / after changing the root directory to newroot, i.e., ",
+		"inside the chroot. This option is only permitted when newroot is the old / directory."))
+	flag.Parse()
+}
+
+func stringToUint32(str string) (uint32, error) {
+	ret, err := strconv.ParseUint(str, 10, 32)
+	if err != nil {
+		return 0, err
+	}
+	return uint32(ret), nil
+}
+
+func getCurrentUser() (uid uint32, gid uint32) {
+	return uint32(syscall.Geteuid()), uint32(syscall.Getegid())
+}
+
+func parseUserspec() (uid uint32, gid uint32, err error) {
+	userspecSplit := strings.Split(userspecFlag, ":")
+	if len(userspecSplit) != 2 || userspecSplit[1] == "" {
+		return 0, 0, fmt.Errorf("Syntax error")
```

Comment:
Perhaps include more information, something like `expected user spec flag to be ":" separated values received %s, ` and maybe print usage.